cocoa/src/toga_cocoa/app.py

```diff
@@ -38,11 +39,26 @@
 class AppDelegate(NSObject):
     interface = objc_property(object, weak=True)
     impl = objc_property(object, weak=True)
+    windows_visible_before_app_hide = []
 
     @objc_method
     def applicationDidFinishLaunching_(self, notification):
         self.native.activateIgnoringOtherApps(True)
 
+    @objc_method
+    def applicationWillHide_(self, notification):
+        for window in self.interface.windows:
+            # on_hide() is triggered only on windows which are in
+            # visible-to-user (i.e., not in minimized or hidden).
+            if window.visible and window.state != WindowState.MINIMIZED:
+                window.on_hide()
+
+    @objc_method
+    def applicationDidUnhide_(self, notification):
+        for window in self.interface.windows:
+            if window.visible:
```

Comment:
I'm surprised by the lack of symmetry here. Either this doesn't *also* need a "if != MINIMIZED" check, or the MINIMIZED check on `WillHide` seems redundant.